### ISSUE TITLE
native app root.location.protocol undefined

### ISSUE THREAD

**Issue by AmareshB** [keen/keen-analysis.js#28]
I am trying to integrate keen with react-native app.
But I am not able to access any methods.
`import Keen from 'keen-analysis';`

`var client = new Keen({
    projectId: 'dummy',
    writeKey: 'dummy'
  });
console.log(client);`

client object gets consoled.

```
client
  .query('count', {
    event_collection: 'sampleData',
    timeframe: 'this_14_days'
  })
  .then(res => {
    // Handle results
    console.log("queryResult::",res );
  })
  .catch(err => {
    // Handle errors
    console.log("err::",err);
  });
```

But this gives error as : 
 `err:: [TypeError: undefined is not an object (evaluating 'root.location.protocol')]`

Need help on this.

**Comment by fieg** [comment]
Do you happen to use Safari? I have a similar problem and think it also has to do with the client: https://github.com/keen/keen-dataviz.js/issues/93

**Comment by AmareshB** [comment]
@fieg I am using it on ios/android emulators. 
So browser is out of the picture here.

**Comment by fieg** [comment]
FYI my problem was caused due to an adblocker.

**Comment by adamkasprowicz** [comment]
it looks like your emulator is not having window.location object
similar problem to https://github.com/mulesoft/js-client-oauth2/issues/49#issuecomment-303805976
try
`window.location = { protocol: 'https:'}`

**Comment by adamkasprowicz** [comment]
Fixed in keen-analysis@2.0.0
